index.js

```diff
@@ -18,14 +18,54 @@ const { socketYmlSchemaV1 } = require('./lib/v1')
  * @property {boolean} [authenticatedProjectReports] enable/disable authenticated project report URLs
  */
 
+/**
+ * Configures how an issue should be handled by a given layer of configuration.
+ * - "defer" will cause the issue to bubble up to a higher layer. E.G. socket.yml may defer to organization settings. This acts as if the value is unset.
+ * - "error" will cause an alert to be shown and require some kind of interaction/setting and will block usage if it is encountered. The interaction may be a prompt, a PR comment, etc.
+ * - "warn" will cause an alert to be shown but will not require any interaction/setting and will not block usage if it is encountered.
+ * - "ignore" will cause an issue to not be shown and not require any interaction/setting and will not block usage if it is encountered.
+ *
+ * @typedef {'defer' | 'error' | 'warn' | 'ignore'} SocketIssueRuleAction
+ */
+
+/**
+ * @typedef SocketIssueRuleObject
+ * @property {SocketIssueRuleAction} [action]
+ */
+
+/**
+ * @typedef {boolean | SocketIssueRuleObject} SocketIssueRuleValue
```

Comment:
A few thoughts after mulling over it. Curious what your thoughts are:

Did you consider something more like

```suggestion
 * @typedef {boolean | 'defer' | 'error' | 'warn' | 'ignore'} SocketIssueRuleValue
```

Just thinking 

```yaml
issueRules:
  - didYouMean: 'warn'
  - installScript: 'error'
```

Is easier to remember than

```yaml
  - didYouMean: 
       action: 'warn'
  - installScript: 
       action: 'error'
```

If we have plans for additional keys, maybe the object makes sense. Do we have an idea what those would be and how they would be used?

The other thought I had was what was the purpose of the `defer` rule action? `error` and `warn` take the place of `true`, `ignore` is a synonym for `false`. But omitting a rule is implicitly a `defer`. Does it enable some kind of new functionality or just serve the role of explicitly defining that behavior?

Reply:
We definitely should look at telemetry and role based prompting here which is why I'm not keen on a string, if we do a string now it would be more breaking in the future. `defer` is to allow explicit saving of that so that the behavior is recorded and not changed. Some UI toggling for example of `true`/`false` with a checkbox cannot represent this currently and has an implicit (very hard to understand) 3rd state that is like defer going on right now.